File: src/Nebula/Modules/InvoiceHub/Dto/EmpresaHub.cs
using System.Text.Json.Serialization;

namespace Nebula.Modules.InvoiceHub.Dto;

public class EmpresaHub
{
    [JsonPropertyName("ruc")]
    public string Ruc { get; set; } = string.Empty;

    [JsonPropertyName("companyId")]
    public string CompanyId { get; set; } = string.Empty;

    [JsonPropertyName("razonSocial")]
    public string RazonSocial { get; set; } = string.Empty;

    [JsonPropertyName("nombreComercial")]
    public string NombreComercial { get; set; } = string.Empty;

    [JsonPropertyName("sunatEndpoint")]
    public string SunatEndpoint { get; set; } = string.Empty;

    [JsonPropertyName("claveSol")]
    public ClaveSolHub ClaveSol { get; set; } = new ClaveSolHub();

    [JsonPropertyName("address")]
    public AddressHub Address { get; set; } = new AddressHub();
}

File: src/xml-sunat/src/Classes/EmpresaHub.php
<?php

namespace App\Classes;

class EmpresaHub
{
    public ?string $ruc = null;
    public ?string $companyId = null;
    public ?string $razonSocial = null;
    public ?string $nombreComercial = null;
    public ?string $sunatEndpoint = null;
    public ?ClaveSolHub $claveSol = null;
    public ?AddressHub $address = null;

    public static function createFromJson(string $json): self
    {
        $data = json_decode($json, true);

        $empresaDto = new self();
        $empresaDto->ruc = $data['ruc'] ?? null;
        $empresaDto->companyId = $data['companyId'] ?? null;
        $empresaDto->razonSocial = $data['razonSocial'] ?? null;
        $empresaDto->nombreComercial = $data['nombreComercial'] ?? null;
        $empresaDto->sunatEndpoint = $data['sunatEndpoint'] ?? null;

        if (isset($data['claveSol'])) {
            $empresaDto->claveSol = ClaveSolHub::createFromJson(json_encode($data['claveSol']));
        }

        if (isset($data['address'])) {
            $empresaDto->address = AddressHub::createFromJson(json_encode($data['address']));
        }

        return $empresaDto;
    }
}